src/filesystem/unix/SDL_sysfilesystem.c

```diff
@@ -172,13 +172,31 @@ SDL_GetBasePath(void)
         sysctl(mib, 4, cmdline, &len, NULL, 0);
 
         exe = cmdline[0];
+        char *pwddst = NULL;
         if (SDL_strchr(exe, '/') == NULL) {  /* not a relative or absolute path, check $PATH for it */
             exe = search_path_for_binary(cmdline[0]);
+        } else {
+            if (exe && *exe == '.') {
+                const char *pwd = SDL_getenv("PWD");
+                if (pwd && *pwd) {
+                    pwddst = (char *) SDL_malloc(PATH_MAX + 1);
+                    strcpy(pwddst, pwd);
+                    strcat(pwddst, "/");
+                    strcat(pwddst, exe);
```

Comment:
This seems very unsafe as it could easily end up writing out of bounds. Maybe try `SDL_asprintf`?